packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx

```diff
@@ -0,0 +1,193 @@
+import Button, { BtnSizes } from '@celo/react-components/components/Button.v2'
+import KeyboardAwareScrollView from '@celo/react-components/components/KeyboardAwareScrollView'
+import KeyboardSpacer from '@celo/react-components/components/KeyboardSpacer'
+import colors from '@celo/react-components/styles/colors.v2'
+import fontStyles from '@celo/react-components/styles/fonts.v2'
+import variables from '@celo/react-components/styles/variables'
+import { parseInputAmount } from '@celo/utils/src/parsing'
+import { StackScreenProps } from '@react-navigation/stack'
+import BigNumber from 'bignumber.js'
+import * as React from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import { Platform, StyleSheet, Text, TextInput, View } from 'react-native'
+import { getNumberFormatSettings } from 'react-native-localize'
+import SafeAreaView from 'react-native-safe-area-view'
+import { useDispatch, useSelector } from 'react-redux'
+import CurrencyDisplay from 'src/components/CurrencyDisplay'
+import LineItemRow from 'src/components/LineItemRow'
+import { DOLLAR_ADD_FUNDS_MIN_AMOUNT, DOLLAR_CASH_OUT_MIN_AMOUNT } from 'src/config'
+import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
+import { Namespaces } from 'src/i18n'
+import {
+  convertDollarsToMaxSupportedPrecision,
+  convertLocalAmountToDollars,
+} from 'src/localCurrency/convert'
+import { useLocalCurrencyCode } from 'src/localCurrency/hooks'
+import { getLocalCurrencyExchangeRate, getLocalCurrencySymbol } from 'src/localCurrency/selectors'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { getRecentPayments } from 'src/send/selectors'
+import { isPaymentLimitReached, showLimitReachedError } from 'src/send/utils'
+import DisconnectBanner from 'src/shared/DisconnectBanner'
+import { dollarBalanceSelector } from 'src/stableToken/selectors'
+
+const { decimalSeparator } = getNumberFormatSettings()
+
+type RouteProps = StackScreenProps<StackParamList, Screens.FiatExchangeAmount>
+
+type Props = RouteProps
+
+const oneDollarAmount = {
+  value: new BigNumber('1'),
+  currencyCode: CURRENCIES[CURRENCY_ENUM.DOLLAR].code,
+}
+
+export function ExchangeTradeScreen({ navigation, route }: Props) {
+  function isNextButtonValid() {
+    const passMinAmount = dollarAmount.isGreaterThanOrEqualTo(
```

Comment:
I find it strange that you can reference `dollarAmount` before its been initialized. JS is strange.

Reply:
Yeah, quite often I think the same way and go back to https://github.com/getify/You-Dont-Know-JS/blob/1st-ed/scope%20%26%20closures/ch4.md to remind myself 🙂